packages/opentelemetry-exporter-prometheus/src/PrometheusSerializer.ts

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { MetricRecord, AggregatorKind } from '@opentelemetry/metrics';
+import { PrometheusExportType, PrometheusCheckpoint } from './types';
+import { Labels } from '@opentelemetry/api';
+import { hrTimeToMilliseconds } from '@opentelemetry/core';
+
+function escapeString(str: string) {
+  return str.replace(/\n/g, '\\n').replace(/\\(?!n)/g, '\\\\');
+}
+
+function escapeLabelValue(str: string) {
+  if (typeof str !== 'string') {
+    str = String(str);
+  }
+  return escapeString(str).replace(/"/g, '\\"');
+}
+
+const invalidCharacterRegex = /[^a-z0-9_]/gi;
+/**
+ * Ensures metric names are valid Prometheus metric names by removing
+ * characters allowed by OpenTelemetry but disallowed by Prometheus.
+ *
+ * https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
+ *
+ * 1. Names must match `[a-zA-Z_:][a-zA-Z0-9_:]*`
+ *
+ * 2. Colons are reserved for user defined recording rules.
+ * They should not be used by exporters or direct instrumentation.
+ *
+ * OpenTelemetry metric names are already validated in the Meter when they are created,
+ * and they match the format `[a-zA-Z][a-zA-Z0-9_.\-]*` which is very close to a valid
+ * prometheus metric name, so we only need to strip characters valid in OpenTelemetry
+ * but not valid in prometheus and replace them with '_'.
+ *
+ * @param name name to be sanitized
+ */
+function sanitizePrometheusMetricName(name: string): string {
+  return name.replace(invalidCharacterRegex, '_'); // replace all invalid characters with '_'
+}
+
+function valueString(value: number) {
+  if (Number.isNaN(value)) {
+    return 'Nan';
+  } else if (!Number.isFinite(value)) {
```

Comment:
this is wrong - completely opposite

Reply:
I don't get what you mean by 'completely opposite'. This function serializes the metric values and this branch converts `+Infinity` and `-Infinity` (those not finite numbers) to '-Inf' and '+Inf' respectively. 

Can you elaborate on your opinion?

Reply:
you have exclamation mark in front of your check

Reply:
We do testing this here https://github.com/open-telemetry/opentelemetry-js/pull/1310/files#diff-6db9781fe72458cffe764ca6261f6926R407. I still don't quite understand your statement. As the exclamation mark is intentional to only format non-finite values like `+Inifinite` and `-Infinite` to `+Inf` and `-Inf` respectively.